### PR TITLE
FIX: Ensure templateflow Layout is re-indexed when updating

### LINKED ISSUE #58: TemplateFlow should re-index the BIDS layout after update
Currently, when `TEMPLATEFLOW_HOME` has been updated in the same python session, the index of the archive remains out-of-date.

### PR DESCRIPTION
**Pull request by mgxd**
Closes #58 

~~Needs a unittest.~~

Also fixes the failing unittest, and catches a bug where `allow_redirect` was not being passed onto certain requests.

### CODE REVIEW

**Comment by mgxd** [comment]
@oesteban pinging here

File: templateflow/tests/test_api.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
     "\tauthor = {Vladimir Fonov and Alan C. Evans and Kelly Botteron and C. "
     "Robert Almli and Robert C. McKinstry and D. Louis Collins},\n"
     "\ttitle = {Unbiased average age-appropriate atlases for pediatric studies},\n"
-    "\tjournal = {{NeuroImage}}\n}"
+    "\tjournal = {NeuroImage}}\n}"
```
Comment (oesteban) [inline templateflow/tests/test_api.py:27]: why this?

**Review by oesteban — APPROVED** [review]

File: templateflow/tests/test_api.py (hunk 1 above)
Comment (mgxd) [reply templateflow/tests/test_api.py:27]: i believe it was a bug in earlier versions of `doi2bib` (fixes tests when using 0.4)

Comment (oesteban) [reply templateflow/tests/test_api.py:27]: seems imbalanced - the other examples don't have the second closing curly brace

Comment (mgxd) [reply templateflow/tests/test_api.py:27]: opened an issue at https://github.com/bibcure/doi2bib/issues/6

File: setup.cfg [hunk 2]
```diff
@@ -49,7 +49,7 @@ exclude =
 
 [options.extras_require]
 citations =
-    doi2bib
+    doi2bib >= 0.4.0
```
Comment (oesteban) [inline setup.cfg:52]: Didn't you mean

```suggestion
    doi2bib < 0.4.0
```

?

**Pull request merged**